contracts/SmartLoan.sol

```diff
@@ -0,0 +1,459 @@
+    IERC20Metadata token = getERC20TokenInstance(asset);
+    uint256 balance = token.balanceOf(address(this));
+    if (balance > 0) {
+      uint256 minSaleAmount = exchange.getMinimumERC20TokenAmountForExactAVAX(targetAvaxAmount, supportedAssets.getAssetAddress(asset));
+      if (balance < minSaleAmount) {
+        uint256 saleAvaxValue = exchange.getEstimatedAVAXFromERC20Token(balance, supportedAssets.getAssetAddress(asset));
+        nonSolventAssetSale(asset, balance, saleAvaxValue);
+      } else {
+        nonSolventAssetSale(asset, minSaleAmount, targetAvaxAmount);
+      }
+    }
+  }
+
+
+  /**
+  * This function attempts to repay the _repayAmount back to the pool as well as pay a bonus to liquidator.
+  * If there is not enough AVAX balance to repay the _repayAmount then the available AVAX balance will be repaid and no
+  * liquidation bonus will be paid to the liquidator.
+  **/
+  function attemptRepay(uint256 _repayAmount) internal {
+    if (address(this).balance <= _repayAmount) {
+      repay(address(this).balance);
+    } else {
+      repay(_repayAmount);
+    }
+  }
+
+
+  function payBonus(uint256 _bonus) internal {
+    if (_bonus < address(this).balance) {
+      payable(msg.sender).transfer(_bonus);
+    } else if (address(this).balance != 0) {
+      payable(msg.sender).transfer(address(this).balance);
+    }
+  }
+
+
+  function selloutLoan() external onlyOwner successfullSellout {
+    bytes32[] memory assets = supportedAssets.getAllAssets();
+    for (uint i = 0; i < assets.length; i++) {
+      nonSolventAssetSale(assets[i], getERC20TokenInstance(assets[i]).balanceOf(address(this)), 0);
+    }
+
+    uint256 debt = getDebt();
+    require(address(this).balance >= debt, "Selling out all assets without repaying the whole debt is not allowed.");
+    repay(debt);
+    if (address(this).balance > 0) {
+      withdraw(address(this).balance);
+    }
+  }
+
+
+  function selloutInsolventLoan(uint256 repayAmount) external successfullSellout {
+    require(!isSolvent(), "Cannot sellout a solvent account");
+
+    uint256 debt = getDebt();
+    if (repayAmount > debt) {
+      repayAmount = debt;
+    }
+    uint256 bonus = repayAmount * LIQUIDATION_BONUS / PERCENTAGE_PRECISION;
+    uint256 totalRepayAmount = repayAmount + bonus;
+
+    sellout(totalRepayAmount);
+    attemptRepay(repayAmount);
+    payBonus(bonus);
+  }
+
+
+  /**
+  * This function role is to sell part/all of the available assets in order to bring the loan back to a solvent state.
+  *
+  **/
+  function sellout(uint256 totalRepayAmount) private {
+    if (address(this).balance < (totalRepayAmount)) {
+      bytes32[] memory assets = supportedAssets.getAllAssets();
+      for (uint i = 0; i < assets.length; i++) {
+        nonSolventPartialOrFullAssetSale(assets[i], totalRepayAmount - address(this).balance);
+        if (address(this).balance >= totalRepayAmount) {
```

Comment:
If we move it to the first line in the loop we won't need the if before.

Reply:
And the code will be cleaner

Reply:
Good catch. Applied :)